### PR TITLE
Make tests reliable around osx ping6 bug.

### LINKED ISSUE #19435: Failure of .UnixPingUtilityTests.PacketSizeIsRespected on OSX
```
System.Net.NetworkInformation.Tests.UnixPingUtilityTests.PacketSizeIsRespected(payloadSize: 50) [FAIL]
        System.Exception : Ping output was <PING6(98=40+8+50 bytes) ::1 --> ::1
        >
        ---- System.ArgumentOutOfRangeException : Index was out of range. Must be non-negative and less than the size of the collection.
        Parameter name: startIndex
        Stack Trace:
           /Users/dotnet-bot/j/workspace/dotnet_corefx/master/osx10.12_debug_prtest/src/System.Net.Ping/tests/FunctionalTests/UnixPingUtilityTests.cs(60,0): at System.Net.NetworkInformation.Tests.UnixPingUtilityTests.<PacketSizeIsRespected>d__1.MoveNext()
           --- End of stack trace from previous location where exception was thrown ---
              at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
              at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
           --- End of stack trace from previous location where exception was thrown ---
              at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
              at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
           --- End of stack trace from previous location where exception was thrown ---
              at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
              at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
           ----- Inner Stack Trace -----
              at System.String.LastIndexOf(String value, Int32 startIndex, Int32 count, StringComparison comparisonType)
           /Users/dotnet-bot/j/workspace/dotnet_corefx/master/osx10.12_debug_prtest/src/System.Net.Ping/tests/FunctionalTests/UnixPingUtilityTests.cs(67,0): at System.Net.NetworkInformation.Tests.UnixPingUtilityTests.ParseReturnedPacketSize(String pingOutput)
           /Users/dotnet-bot/j/workspace/dotnet_corefx/master/osx10.12_debug_prtest/src/System.Net.Ping/tests/FunctionalTests/UnixPingUtilityTests.cs(47,0): at System.Net.NetworkInformation.Tests.UnixPingUtilityTests.<PacketSizeIsRespected>d__1.MoveNext()
  Finished:    System.Net.Ping.Functional.Tests
  ```

https://ci.dot.net/job/dotnet_corefx/job/master/job/osx10.12_debug_prtest/3051/consoleText

Compare https://github.com/dotnet/corefx/issues/15018

### PR DESCRIPTION
**Pull request by Priya91**
fixes #19435 

cc @danmosemsft

### CODE REVIEW

**Comment by Priya91** [comment]
@dotnet-bot help

**Comment by Priya91** [comment]
@dotnet-bot test outerloop netcoreapp OSX10.12 Debug
@dotnet-bot test outerloop netcoreapp OSX10.12 Release

**Comment by stephentoub** [comment]
> around osx ping6 bug

It's a known bug in macOS?

File: src/System.Net.Ping/tests/FunctionalTests/UnixPingUtilityTests.cs [hunk 1]
```diff
@@ -39,6 +40,13 @@ public class UnixPingUtilityTests
 
             string pingOutput = p.StandardOutput.ReadToEnd();
             Assert.True(p.WaitForExit(TestSettings.PingTimeout), "Ping process did not exit in " + TestSettings.PingTimeout + " ms.");
+            if (p.ExitCode == 1 || p.ExitCode == 2)
+            {
+                // Workaround known OSX bug in ping6 utility.
+                Assert.Equal(utilityPath, UnixCommandLinePing.Ping6UtilityPath);
```
Comment (danmosemsft) [inline src/System.Net.Ping/tests/FunctionalTests/UnixPingUtilityTests.cs]: does this basically mean we're not testing ping on OSX (or put another way is this a long term change)
the tests aren't doing anything unusual as far as I know - the product would throw PingException but it's still failed from the user's point of view. maybe there should be an issue to investigate more - unless it's fixed by Apple meantime.

**Comment by danmosemsft** [comment]
@stpehentoub I think so. Some history in https://github.com/dotnet/corefx/issues/15018 https://github.com/dotnet/corefx/issues/18125
I think we should do this to stop the failures, but it needs investigation since the product will have hte same issue.

**Comment by stephentoub** [comment]
Ah, right, thanks for the reminder.  Subsequently it'd be good to try to get official confirmation of that, a bug number we could track, etc.

**Comment by Priya91** [comment]
The following ping tests failed in outerloop run:

```
System.Net.NetworkInformation.Tests.PingTest.SendPingAsyncWithIPAddressAndTimeoutAndBufferAndPingOptions_Unix [FAIL]
System.Net.NetworkInformation.Tests.PingTest.SendPingAsyncWithIPAddressAndTimeoutAndBuffer_Unix [FAIL]
System.Net.NetworkInformation.Tests.PingTest.SendPings_ReuseInstance_Hostname [FAIL]
System.Net.NetworkInformation.Tests.PingTest.Sends_ReuseInstance_Hostname [FAIL]
System.Net.NetworkInformation.Tests.PingTest.SendPingAsyncWithIPAddress_AddressAsString [FAIL]
System.Net.NetworkInformation.Tests.PingTest.SendPingAsyncWithIPAddress [FAIL]
```

Will handle these in different PR, after confirming if it's the same OSX bug. @danmosemsft These failures seem to be very recent, did anything change in the OSX OS image, that is causing these tests to fail more frequently?

File: src/System.Net.Ping/tests/FunctionalTests/UnixPingUtilityTests.cs (hunk 1 above)
Comment (Priya91) [reply src/System.Net.Ping/tests/FunctionalTests/UnixPingUtilityTests.cs]: This codepath will not be hit if the osx bug gets fixed. We are only handling the specific bug case.

**Comment by Priya91** [comment]
> Subsequently it'd be good to try to get official confirmation of that, a bug number we could track, etc.

I remember searching for some bug info around this, during the time of the first report of the issue, couldn't find it. Will look for it again.

**Comment by danmosemsft** [comment]
@dotnet/dnceng is this MSBuild failure just a consequence of jenkins terminating the parent msbuild.exe, or is it an actual crash?
It's usually just because of jenkins terminating it, but I'm not sure all the tests ran in this case.
```
  chmod +x /Users/dotnet-bot/j/workspace/dotnet_corefx/master/outerloop_netcoreapp_osx_release_prtest/bin/testhost/netcoreapp-OSX-Release-x64//dotnet
  /Users/dotnet-bot/j/workspace/dotnet_corefx/master/outerloop_netcoreapp_osx_release_prtest/bin/testhost/netcoreapp-OSX-Release-x64//dotnet xunit.console.netcore.exe System.Net.HttpListener.Tests.dll  -xml testResults.xml -notrait category=nonnetcoreapptests -notrait category=nonosxtests  -notrait category=IgnoreForCI -notrait category=failing
  python DumplingHelper.py collect_dump $\ `pwd`  System.Net.HttpListener.Tests /Users/dotnet-bot/j/workspace/dotnet_corefx/master/outerloop_netcoreapp_osx_release_prtest/bin/runtime/netcoreapp-OSX-Release-x64/,/Users/dotnet-bot/j/workspace/dotnet_corefx/master/outerloop_netcoreapp_osx_release_prtest/bin/tests/System.Net.HttpListener.Tests/netcoreapp-OSX-Release-x64/
  /Users/dotnet-bot/j/workspace/dotnet_corefx/master/outerloop_netcoreapp_osx_release_prtest/bin/tests/System.Net.HttpListener.Tests/netcoreapp-OSX-Release-x64 /Users/dotnet-bot/j/workspace/dotnet_corefx/master/outerloop_netcoreapp_osx_release_prtest/src/System.Net.HttpListener/tests
  /tmp/tmp484efda55e2b49d684be5d3d7ce84435.exec.cmd: line 2: 33123 Terminated: 15          /Users/dotnet-bot/j/workspace/dotnet_corefx/master/outerloop_netcoreapp_osx_release_prtest/bin/tests/System.Net.Requests.Tests/netcoreapp-OSX-Release-x64/RunTests.sh /Users/dotnet-bot/j/workspace/dotnet_corefx/master/outerloop_netcoreapp_osx_release_prtest/bin/testhost/netcoreapp-OSX-Release-x64/
./build-tests.sh: line 34: 30166 Terminated: 15          $working_tree_root/run.sh build-managed -testsOnly $__UnprocessedBuildArgs
MSBUILD : error MSB4166: Child node "3" exited prematurely. Shutting down. Diagnostic information may be found in files in the temporary files directory named MSBuild_*.failure.txt.
Agent went offline during the build
```

**Pull request merged**